pkg/api/api.go

```diff
@@ -287,3 +288,15 @@ func RespondError(w http.ResponseWriter, apiErr *ApiError, data interface{}) {
 		Data:      data,
 	})
 }
+
+func warningsToString(errors []error) []string {
+	warningStrings := make([]string, len(errors))
+
+	for i, err := range errors {
+		if err != nil {
+			warningStrings[i] = err.Error()
+		}
+	}
+
+	return warningStrings
+}
```

Comment:
I wonder if it is better to split the changes and we add warnings to backend response first...